packages/storage/test/unit/location.test.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * @license
+ * Copyright 2017 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { expect } from 'chai';
+import { DEFAULT_HOST } from '../../src/implementation/constants';
+import { Location } from '../../src/implementation/location';
+
+describe('Firebase Storage > Location', () => {
+  it('makeFromUrl handles an emulator url correctly', () => {
+    const loc = Location.makeFromUrl(
+      'http://localhost:3001/v0/b/abcdefg.appspot.com/o/abcde.txt',
+      'localhost:3001'
+    );
+    expect(loc.bucket).to.equal('abcdefg.appspot.com');
```

Comment:
Should we test the protocol here as well?

Reply:
Location doesn't seem to store the host or the protocol, makeFromUrl just uses them to do regex matches and then throws them away. It only seems to have bucket, and path (just the part after the bucket).

Reply:
Okay, got it.